File: java/src/main/java/se/gewalli/users/model/V1Mapper.java
package se.gewalli.users.model;

import se.gewalli.users.domain.User;
import se.gewalli.users.domain.UserRole;

import java.util.stream.Collectors;

public class V1Mapper {
    public static V1User map(User value)
    {
        return new V1User()
                .id(value.getId().getId())
                .email(value.getEmail())
                .isActive(value.isActive())
                .name(value.getName())
                .roles(value.getRoles().stream().map(V1Mapper::mapRole).collect(Collectors.toList()));
    }

    private static String mapRole(UserRole arg)
    {
        switch (arg)
        {
            case ADMINISTRATOR: return "A";
            case NORMAL: return "N";
            case SUPPORT: return "S";
            default: throw new RuntimeException(arg.toString());
        }
    }
}

File: csharp/API/V1/V1Mapper.cs
using API.Abstractions;
using Domain;

namespace API.V1;

public class V1Mapper : IMapper<User,V1User>
{
    public static V1User Map(User value)
    {
        return new V1User(
            Id: value.Id.Id, 
            Email: value.Email, 
            IsActive: value.IsActive, 
            Name: value.Name, 
            Roles: value.Roles.Select(MapRole).ToArray());
    }

    private static string MapRole(UserRole arg)
    {
        switch (arg)
        {
            case UserRole.Administrator: return "A";
            case UserRole.Normal: return "N";
            case UserRole.Support: return "S";
            default: throw new Exception(arg.ToString());
        }
    }
}